tracing/opentelemetry/src/main/java/io/helidon/tracing/opentelemetry/HelidonOpenTelemetry.java

```diff
@@ -49,4 +56,51 @@ public static OpenTelemetryTracer create(OpenTelemetry telemetry, Tracer tracer,
     public static io.helidon.tracing.Span create(Span span) {
         return new OpenTelemetrySpan(span);
     }
+
+    /**
+     * Check if OpenTelemetry is present by indirect properties.
+     * This class does best explicit check if OTEL_AGENT_PRESENT_PROPERTY config property is set and uses its
+     * value to set the behaviour of OpenTelemetry producer.
+     *
+     * If the value is not explicitly set, the detector does best effort to estimate indirect means if the agent is present.
+     * This detector may stop working if OTEL changes the indirect indicators.
+     */
+    public static class AgentDetector {
+
+        /**
+         * Check if the OTEL Agent is present.
+         *
+         * @param config Configuration
+         * @return boolean
+         */
+        public static boolean isAgentPresent(Config config) {
+
+            //Explicitly check if agent property is set
+            if (config != null) {
+                Optional<Boolean> agentPresent = config.get(OTEL_AGENT_PRESENT_PROPERTY).asBoolean().asOptional();
+                if (agentPresent.isPresent()) {
+                    return agentPresent.get();
+                }
+            }
+
+            if (checkContext() || checkSystemProperties()){
+                if (LOGGER.isLoggable(System.Logger.Level.INFO)) {
+                    LOGGER.log(System.Logger.Level.INFO, "OpenTelemetry Agent detected");
+                }
+                return true;
+            }
+            return false;
+        }
+
+        private static boolean checkSystemProperties() {
+            return System.getProperties().stringPropertyNames()
+                    .stream()
+                    .anyMatch(property -> property.contains(IO_OPENTELEMETRY_JAVAAGENT));
+        }
+
+        private static boolean checkContext() {
+            return io.opentelemetry.context.Context.current().getClass().getName().contains("agent");
```

Comment:
Context should be imported and used without fully qualified package.

Reply:
Done